apple/swift-docc-render
# Add a sidenav navigator to the documentation pages (WIP behind a feature flag, off by default)

Bug/issue #, if applicable: 89031049

## Summary

This PR introduces a few new UI changes, that are scoped only to the documentation pages. The features are all DISABLED, by default, but can be enabled by a feature flag in the theme-settings.json file, under `features.docs.navigator.enable`:

1. New adjustable layout
2. A sidebar tree navigator
3. Updated nav layout and breadcrumbs logic

This work covers both of the proposals:

https://forums.swift.org/t/new-swift-docc-content-styling/55326
https://forums.swift.org/t/swift-docc-sidebar/55250

## Dependencies

https://github.com/apple/swift-docc/pull/84

## Testing

Make sure you download and unzip the `.doccarchive` - [SlothCreator.doccarchive.zip](https://github.com/apple/swift-docc-render/files/8082216/SlothCreator.doccarchive.1.zip)


Steps:
1. run `VUE_APP_DEV_SERVER_PROXY=/path/to/SlothCreator.doccarchive npm run serve`
2. Visit http://localhost:8080/documentation/slothcreator
3. Assert that a sidebar tree navigator is rendered.
4. Assert that you can open/close items and click on the links to navigate to a page
5. Assert that navigating across pages does not loose open/closed state. Even on refresh.
5. Assert that going back/forward works and re-highlights the current item you are on, opening all parents so its visible.
6. Assert you can filter on items and its not slowing down the browser even for very broad queries.
8. Assert that the navigation layout does not go outside of the content container
9. Assert that the navigation breadcrumbs reduce in size and collapse, based on screen size.
9. Assert that the content has a maximum width, after which it gets centered (around 1800px)

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
